### PR TITLE
Expand testing for DependencyBuild

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: pkg/apis/jvmbuildservice/v1alpha1/artifactbuildrequest_types.go [hunk 1]
```diff
@@ -5,18 +5,15 @@ import (
...
-	Message       string `json:"message,omitempty"`
-	RecipeGitHash string `json:"recipeGitHash,omitempty"`
-	SCMURL        string `json:"scmURL,omitempty"`
-	SCMType       string `json:"scmType,omitempty"`
-	Tag           string `json:"tag,omitempty"`
-	Path          string `json:"path,omitempty"`
+	//TODO: conditions?
+	State   string  `json:"state,omitempty"`
+	Message string  `json:"message,omitempty"`
+	ScmInfo SCMInfo `json:"scm,omitempty"`
```
Comment (gabemontero) [inline pkg/apis/jvmbuildservice/v1alpha1/artifactbuildrequest_types.go:16]:
```suggestion
	SCMInfo SCMInfo `json:"scm,omitempty"`
```

I think we need to be case consistent here

File: pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go [hunk 2]
```diff
@@ -13,10 +13,7 @@ const (
 )
 
 type DependencyBuildSpec struct {
-	SCMURL  string `json:"scmURL,omitempty"`
-	SCMType string `json:"scmType,omitempty"`
-	Tag     string `json:"tag,omitempty"`
-	Path    string `json:"path,omitempty"`
+	ScmInfo SCMInfo `json:"scmInfo,omitempty,inline"`
```
Comment (gabemontero) [inline pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go:16]:
```suggestion
	SCMInfo SCMInfo `json:",inline"`
```

I've never seen folks inline the way you had it.  Did you try it with just inline first and it did not work for some reason?

**Review by gabemontero — CHANGES_REQUESTED** [review]

File: pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go (hunk 2 above)
Comment (stuartwdouglas) [reply pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go:16]: ah, that would be why it was not working

**Pull request closed without merging**